cpp/src/search/search.cu

```diff
@@ -248,83 +237,115 @@ bool contains_scalar_dispatch::operator()<cudf::dictionary32>(column_view const&
                                  : false;
 }
 
-}  // namespace
-
-namespace detail {
-bool contains(column_view const& col, scalar const& value, rmm::cuda_stream_view stream)
-{
-  if (col.is_empty()) { return false; }
-  if (not value.is_valid(stream)) { return col.has_nulls(); }
-
-  return cudf::type_dispatcher(col.type(), contains_scalar_dispatch{}, col, value, stream);
-}
-
 struct multi_contains_dispatch {
-  template <typename Element>
-  std::unique_ptr<column> operator()(column_view const& haystack,
-                                     column_view const& needles,
-                                     rmm::cuda_stream_view stream,
-                                     rmm::mr::device_memory_resource* mr)
+  template <typename Type>
```

Comment:
Is `Type` more descriptive than `Element`? The latter is pretty standard at this point.

Reply:
From my observation, template arguments typically have the suffix `_Type` or `_t` such as `DType`, `ElementType`, or `size_t`. Sometimes the entire argument is just `T`.
Please correct me if I'm wrong, `Element` by its name sounds like an object to me, not a type.

Reply:
`ElementType` is common and fine too. I think `Element` is just a shortening of that.